rollup.config.js

```diff
@@ -3,24 +3,60 @@ import commonjs from '@rollup/plugin-commonjs'
 import resolve from '@rollup/plugin-node-resolve'
 import {terser} from 'rollup-plugin-terser'
 import visualizer from 'rollup-plugin-visualizer'
+import packageJson from './package.json'
 
 const extensions = ['.js', '.jsx', '.ts', '.tsx']
-
 const formats = ['esm', 'umd']
-
-const plugins = [
-  babel({extensions, exclude: 'node_modules/**', runtimeHelpers: true}),
-  resolve({extensions}),
-  commonjs(),
-  terser(),
-  visualizer({sourcemap: true})
+const external = [
+  ...Object.keys(packageJson.peerDependencies),
+  ...Object.keys(packageJson.dependencies),
+  ...Object.keys(packageJson.devDependencies)
 ]
 
+const baseConfig = {
+  input: 'src/index.ts',
+  external: id => {
+    return external.some(pkg => {
+      return id.startsWith(pkg)
+    })
+  },
+  plugins: [
+    resolve({
+      extensions
+    }),
+    commonjs(),
+    babel({
```

Comment:
I'd recommend avoiding babel here. We dropped babel from dotcom and all public repos a few years ago because:

- babel is quite slow. Typescript or esbuild or swc are much faster and equally as capable
- Babel tends to err on the side of compatibility which means transforming things that don't necessarily need to be transformed due to edge cases. (An example of this is object rest spread gets transformed if you specify edgehtml as a browser, even though edge supports object rest spread). This results in bigger bundle sizes than necessary.
- Configuring babel can be tricky and can result in varying compatibilities. Typescripts `target` is a little more blunt (just specifying a year to compile down to) but that's much easier to propagate across many repos.

In addition, moving from babel would be a great step to coalescing all our repositories around a central build config. As mentioned doctor uses swc, while all other repositories just use typescript. 

Here's Primer ViewComponents configuration: https://github.com/primer/view_components/blob/b0dbefb471d8262a2a45235ccaf0501260f00cd4/rollup.config.js#L15